llm-chain-openai/src/chatgpt/step.rs

```diff
@@ -50,25 +50,38 @@ impl ToString for Model {
 /// let model = Model::ChatGPT3_5Turbo;
 /// let prompt = ChatPromptTemplate::system_and_user("You are an assistant that speaks like Shakespeare.", "tell me a joke");
 ///
-/// let chat_gpt_step = Step::new(model, prompt);
+/// let chat_gpt_step = Step::new(model, prompt, None);
 /// ```
 #[cfg_attr(feature = "serialization", derive(Serialize, Deserialize))]
 pub struct Step {
     pub(crate) model: Model,
     pub(crate) prompt: ChatPromptTemplate,
+    pub(crate) stream: Option<bool>,
 }
+
 impl Step {
     #[deprecated(since = "0.6.0", note = "Use `for_prompt_and_model` instead")]
-    pub fn new<P: Into<ChatPromptTemplate>>(model: Model, prompt: P) -> Step {
+    pub fn new<P: Into<ChatPromptTemplate>>(model: Model, prompt: P, stream: Option<bool>) -> Step {
         let prompt = prompt.into();
-        Step { model, prompt }
+        Step {
+            model,
+            prompt,
+            stream,
+        }
     }
-    pub fn for_prompt_and_model<P: prompt::Prompt>(model: Model, prompt: P) -> Step {
+    pub fn for_prompt_and_model<P: prompt::Prompt>(
```

Comment:
Maybe we can make it `$methodname_with_streaming` for the streaming methods. We will switch a builder pattern soon